app/main/views/g_cloud.py

```diff
@@ -487,11 +487,18 @@ def view_project(framework_family, project_id):
         current_project_stage = 'search_ended'
     else:
         current_project_stage = 'save_and_refine_search'
+    
+    # custom dimension for google analytics
+    custom_dimensions = [{
```

Comment:
If we're making this something that we can add on the Python side, it would be good to maybe have a `CustomDimension` class in the fe-toolkit we can use to have some sort of standard format for passing into the templates. Probably just a simple object with an ID and value is sufficient, but would be useful.

Reply:
Yep, kinda feel like we don't have anywhere obvious to put (or look for) this right now. I suppose it would have to go into dm-utils.

Perhaps this is only worth putting in if we come across another place where we actually need extra custom dimensions, it's maybe overkill for now given the single place we ever use it.

Reply:
FWIW I *do* think we should consider having a Python library that goes along with the frontend toolkit. I think there are probably quite a few things like the forms stuff that @lfdebrux and you @samuelhwilliams have been doing that might be better separated off from the rest of dm-utils.

But @risicle will hate me for ever for suggesting making another library, so let's not do that just yet!...

Reply:
If you really want to set up another `scripts/` directory, another pytest setup, travis, jenkins job... and keep them all up to date with the way we do these things elsewhere, be my guest.

Reply:
Can we not just have the base template(s) able to consume a `custom_dimensions` dict of key/values to turn into custom dimensions on rendering? a la flash messages?

Reply:
Well yes that's the current situation, and I think we'll leave it that way until this becomes more widespread and if people feel they a class would help ensure we get the right structure everywhere.

It's true that Flash messages in Flask don't use a class, but they do have a special function to call whose argument list determines what structure you can give them, so it's kinda equivalent.